uspv/header.go

```diff
@@ -50,14 +37,11 @@ func checkProofOfWork(header wire.BlockHeader, p *coinparam.Params) bool {
 	}
 
 	// The header hash must be less than the claimed target in the header.
-	var blockHash chainhash.Hash
+  
+    var buf bytes.Buffer
+	_ = wire.WriteBlockHeader(&buf, 0, &header)
```

Comment:
I get an error here; did you also change adiabat/btcd? could you add the PR there and I'll merge that first?